src/main/java/org/apache/maven/shared/incremental/IncrementalBuildHelperRequest.java

```diff
@@ -73,4 +75,20 @@ public IncrementalBuildHelperRequest outputDirectory( File outputDirectory )
         this.outputDirectory = outputDirectory;
         return this;
     }
+
+    public File getGeneratedSourcesDirectory()
+    {
+        return generatedSourcesDirectory;
+    }
+
+    public void setGeneratedSourcesDirectory( File generatedSourcesDirectory )
+    {
+        this.generatedSourcesDirectory = generatedSourcesDirectory;
```

Comment:
Is it worth checking here that the file exists and is a directory? Throw an IllegalArgumentException if not.